addons/calendar/models/calendar_event.py

```diff
@@ -1407,3 +1408,11 @@ def _get_public_fields(self):
             'id', 'active', 'allday',
             'duration', 'user_id', 'interval', 'partner_id',
             'count', 'rrule', 'recurrence_id', 'show_as', 'privacy'}
+
+    @api.model
+    def get_default_duration(self):
+        first_val = self.env['ir.default'].sudo()._get('calendar.event', 'duration', user_id=True, company_id=True)
+        second_val = self.env['ir.default'].sudo()._get('calendar.event', 'duration', user_id=True)
+        third_val = self.env['ir.default'].sudo()._get('calendar.event', 'duration', company_id=True)
+        default_val = self.env['ir.default'].sudo()._get('calendar.event', 'duration') or 1
+        return first_val or second_val or third_val or default_val
```

Comment:
```suggestion
    def get_default_duration(self):
        res = self.env['ir.default'].sudo()._get('calendar.event', 'duration', user_id=True, company_id=True)
        res = res or self.env['ir.default'].sudo()._get('calendar.event', 'duration', user_id=True)
        res = res or self.env['ir.default'].sudo()._get('calendar.event', 'duration', company_id=True)
        res = res or self.env['ir.default'].sudo()._get('calendar.event', 'duration')
        return res or 1
```

This avoid doing extra work when you already found the 1st value